Adding a dropdown box for resources section in the nav-bar

floydtree opened the pull request:
1. Adding a dropdown box to collect all the resources offered in the schema server.
2. Removing the old guidelines page, adding references to the Understanding OCSF doc and Contribution guide.


![Screenshot 2023-11-08 at 12 17 47 PM](https://github.com/ocsf/ocsf-server/assets/89877409/14709ceb-cabe-4680-a81d-906d0856991f)

mikeradka commented:
Taking a look. Going to try the `gh` tool to check out the PR and play around with the new feature a bit.

mikeradka commented on lib/schema_web/templates/layout/app.html.eex:
```diff
@@ -204,25 +195,27 @@ limitations under the License.
+        <div class="dropdown">
+          Resources
+          <div class="dropdown-content">
+            <a class="nav-link" href="https://github.com/ocsf/ocsf-docs/blob/main/Understanding%20OCSF.pdf">Understanding OCSF</a>
```
Might just be a nitpick, but could we have `target="_blank"` to any resources that are not on `schema.ocsf.io`?

floydtree replied:
good catch! adding it in

mikeradka reviewed (approved):
Tested out locally from various views, working as expected.

rmouritzen-splunk commented:
I approved, however I'd like to suggest a change, and if you agree, perhaps implement it with this PR. 

Of the 4 items in the "Resources" dropdown, 3 are links that open a new page, however "Data Types" navigates the schema browser to the its own "Data Types" page. I can see why it was done this, way, however I think "Data Types" should be moved to a sibling of the other top level navigation items. Something like this:
```
Categories   Classes   Base Event   Dictionary   Objects   Data Types   |   Resources
```

rmouritzen-splunk reviewed (changes requested):
Version needs to be bumped from `2.60.0` to `2.61.0` in the top level `mix.exs` file.

floydtree commented:
> ~I approved~, however I'd like to suggest a change, and if you agree, perhaps implement it with this PR.
> 
> _Edit: We need to bump the version, so I changed the approval to a change request. The bit below remains a suggestion to consider._
> 
> Of the 4 items in the "Resources" dropdown, 3 are links that open a new page, however "Data Types" navigates the schema browser to the its own "Data Types" page. I can see why it was done this, way, however I think "Data Types" should be moved to a sibling of the other top level navigation items. Something like this:
> 
> ```
> Categories   Classes   Base Event   Dictionary   Objects   Data Types   |   Resources
> ```



> ~I approved~, however I'd like to suggest a change, and if you agree, perhaps implement it with this PR.
> 
> _Edit: We need to bump the version, so I changed the approval to a change request. The bit below remains a suggestion to consider._
> 
> Of the 4 items in the "Resources" dropdown, 3 are links that open a new page, however "Data Types" navigates the schema browser to the its own "Data Types" page. I can see why it was done this, way, however I think "Data Types" should be moved to a sibling of the other top level navigation items. Something like this:
> 
> ```
> Categories   Classes   Base Event   Dictionary   Objects   Data Types   |   Resources
> ```

Ah the version number, pushed a commit for that. About the the dropdown - Yeah I had initially considered the same style, but later arrived at this since Data Type really fits as a resource for the framework. But I am not married to that idea, I would be okay either way.

rmouritzen-splunk commented:
Approved. In the interest of time, let's live with this for a bit and decide later if "Data Types" should be moved.
